docs/source/protocol_v3.rst

```diff
@@ -1,51 +1,437 @@
+the `Handshake`_ and `Session info`_ packets. 
+
+If the simulation engine does not 
+receive this packet within 1 second of sending the handshake and session info 
+packets, it should assume the receiver is incompatible. Whether the simulation engine
+exits or accepts another connection after this is an implementation decision.
+
+.. code-block:: none
+
+   Header:
+      3 (int32) Go
+      <val> (no type) Unused slot, any value acceptable
+
+
+Handshake
+^^^^^^^^^
+
+Sent from the simulation engine to the receiver after a socket connection
+is established. Unlike other header packets, the last four bytes of this packet are provided in 
+the native endianness of the sending simulation engine's hardware.
+
+The receiver can use this packet to determine both the IMD version
+of the session and the endianness of the simulation engine. By providing 
+the endianness of the machine running the simulation engine, the bulk of the 
+data being sent in the session, i.e. the body packets, do not have to be swapped 
+by the simulation engine before being sent, speeding up execution.
+
+.. code-block:: none
+
+   Header:
+      4 (int32) Handshake
+      3 (int32, unswapped byte order) IMD version used in session
+
+Kill
+^^^^
+
+Sent from the receiver to the simulation engine any time after the `Session info`_
+has been sent to request that the simulation engine
+stops execution of the simulation and exits. Whether or not the simulation engine 
+honors this request is an implementation decision.
+
+.. code-block:: none
+
+   Header:
+      5 (int32) Kill
+      <val> (no type) Unused slot, any value acceptable
+
+MD Communication
+^^^^^^^^^^^^^^^^
+
+Sent from the receiver to the simulation engine any time after the `Session info`_
+has been sent to request that the forces 
+in the body packet are applied to the atoms specified in the body packet. 
+Whether or not the simulation engine honors this request is an implementation decision.
+
+.. code-block:: none
+
+   Header:
+      6 (int32) MD Communication
+      <n_forces> (int32) Number of forces to be applied to the simulation engine
+
+   Body:
+      <array> (int32[n_forces]) Indices of atoms in the IMD system to apply forces to
+      <array> (float32[n_forces * 3]) The X, Y, and Z components of forces to be applied to
+                                      the atoms at the indices specified in the above array
+     
+The array of IMD system indices does not need to be monotonically increasing, meaning 
+the indices can be "out of order". However, indices cannot be duplicated, i.e. 
+the index array cannot contain index "2" twice. In this case, the force vectors should 
+be combined before being sent to the simulation engine to be applied.
+
+Pause
+^^^^^
+
+Sent from the receiver to the simulation engine any time after the `Session info`_
+has been sent to request that the simulation
+engine blocks execution until a `Resume`_ packet is sent. Unlike in IMDv2, pause is idempotent,
```

Comment:
I would segregate all comparisons to v2 into a special section at the end of each package description, maybe a note box. First read all about v3, then about v2. 

or even cleaner: use the reST version added and versionchanged markup!

Reply:
versionchanged/versionadded worked well!